doc/reference/trd-syscalls.md

```diff
@@ -351,14 +351,17 @@ unmodified.
 If the passed callback is not valid (is outside process executable
 memory and is not the Null Callback described below), the kernel MUST
 NOT invoke the requested driver and MUST immediately return a failure
-with a return code of EINVAL.
-
-A passed callback MUST be valid until the next invocation of `subscribe`
-with the same syscall and driver identifier. When userspace invokes
-subscribe, the kernel MUST cancel all pending callbacks for that driver
-and subscribe identifier: it MUST NOT invoke the previous callback after
-the call to subscribe, and MUST NOT invoke the new callback for events
-that occurred before the call to subscribe.
+with a return code of EINVAL. The currently registered callback
+remains unchanged.
+
+A passed callback MUST be valid until the next invocation of
```

Comment:
I think "valid" should be added after "A passed callback MUST be valid until the next", because an existing callback must remain valid after a failed `subscribe` call.

Reply:
I agree. I've actually changed it so say "accepted" instead of "valid", as it not only matters whether the kernel deems this to be a valid callback, but also whether the capsule accepts the subdriver number, and swaps the callbacks internally.